pkg/expression/column.go

```diff
@@ -511,9 +511,13 @@ func (col *Column) Hash64(h base.Hasher) {
 
 // Equals implements HashEquals.<1st> interface.
 func (col *Column) Equals(other any) bool {
-	if other == nil {
-		return false
+	// If col itself is nil, check whether other is also nil
+	if col == nil {
+		// Checking if other is of type *Column and nil
+		col2, ok := other.(*Column)
+		return ok && col2 == nil
 	}
+	// Attempt to convert other to *Column
 	var col2 *Column
 	switch x := other.(type) {
```

Comment:
this kind of way can be eliminated, we can make it only receive *pointer as its argument here. Let the caller to pass &XX if it's a structure.